streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -502,6 +517,13 @@ public StreamsConfig(final Map<?, ?> props) {
         consumerProps.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, originals().get(BOOTSTRAP_SERVERS_CONFIG));
         // remove deprecate ZK config
         consumerProps.remove(ZOOKEEPER_CONNECT_CONFIG);
+        if (eosEnabled) {
+            if (clientProvidedProps.containsKey(ConsumerConfig.ISOLATION_LEVEL_CONFIG)) {
```

Comment:
IMHO, Streams is more than a client. Thus, it seems ok to behave differently. \cc @guozhangwang @dguy @enothereska @miguno @sriramsub